src/devices/Dhtxx/DhtSensor.cs

```diff
@@ -0,0 +1,307 @@
+                else
+                    return Double.MaxValue;
+            }
+        }
+
+        /// <summary>
+        /// Get the temperature in Celsius
+        /// </summary>
+        /// <param name="temperatureInCelsius">The temperature in Celsius</param>
+        /// <returns>Returns <c>true</c> if the read is successful</returns>
+        /// <remarks>
+        /// If last read was not successfull, it returns double.MaxValue
+        /// </remarks>
+        public bool TryGetTemperatureInCelsius(out double temperatureInCelsius)
+        {
+            var ret = ReadData();
+            temperatureInCelsius = TemperatureInCelsius;
+            return ret;
+        }
+
+        /// <summary>
+        /// Get the temperature in Farenheit
+        /// </summary>
+        /// <param name="temperatureInFarenheit">The temperature in Farenheit</param>
+        /// <returns>Returns <c>true</c> if the read is successful</returns>
+        /// <remarks>
+        /// If last read was not successfull, it returns double.MaxValue
+        /// </remarks>
+        public bool TryGetTemperatureInFarenheit(out double temperatureInFarenheit)
+        {
+            var ret = ReadData();
+            temperatureInFarenheit = TemperatureInFarenheit;
+            return ret;
+        }
+
+        /// <summary>
+        /// Get the last read humidity
+        /// </summary>
+        /// <remarks>
+        /// If last read was not successfull, it returns double.MaxValue
+        /// </remarks>
+        public double Humidity
+        {
+            get
+            {
+                if (_dhtType == DhtType.Dht11)
+                    return GetHumidityDht11();
+                else
+                    return GetHumidityDht22();
+            }
+        }
+
+        /// <summary>
+        /// Get the relative humidity in the air
+        /// </summary>
+        /// <param name="humidity">The percentage of humidity in the air</param>
+        /// <returns>Returns <c>true</c> if the read is successful</returns>
+        /// <remarks>
+        /// If last read was not successfull, it returns double.MaxValue
+        /// </remarks>
+        public bool TryGetHumidity(out double humidity)
+        {
+            var ret = ReadData();
+            humidity = Humidity;
+            return ret;
+        }
+
+        /// <summary>
+        /// Get the temperature in Celsius and the relative humidity in the air
+        /// </summary>
+        /// <param name="temperatureInCelsius">The temperature in Celsius</param>
+        /// <param name="humidity">The percentage of humidity in the air</param>
+        /// <returns>Returns <c>true</c> if the read is successful</returns>
+        /// <remarks>
+        /// If last read was not successfull, it returns double.MaxValue
+        /// </remarks>
+        public bool TryGetTemperatureInCelsiusHumidity(out double temperatureInCelsius, out double humidity)
```

Comment:
I think environment sensors are largely standardized on Celsius.  Why not drop Celcius from the API name (keep it in the out parameter name); and then provide a simple C->F F->C converter elsewhere instead of having all the Fahrenheit overloads?  just a thought.